design/multiple-listeners-design.md

```diff
@@ -0,0 +1,111 @@
+# Automatic redirection with arbitrary listeners
+
+Status: Accepted
```

Comment:
As we've done for other current designs, setting this to `Accepted` because it won't be merged unless it is accepted.